pkg/kubelet/kubelet_node_status.go

```diff
@@ -370,13 +372,21 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 	// apiserver cache (the data might be slightly delayed but it doesn't
 	// seem to cause more conflict - the delays are pretty small).
 	// If it result in a conflict, all retries are served directly from etcd.
-	opts := metav1.GetOptions{}
+	var node *v1.Node
+	var err error
 	if tryNumber == 0 {
-		util.FromApiserverCache(&opts)
-	}
-	node, err := kl.heartbeatClient.Nodes().Get(string(kl.nodeName), opts)
-	if err != nil {
-		return fmt.Errorf("error getting node %q: %v", kl.nodeName, err)
+		node, err = kl.nodeInfo.GetNodeInfo(string(kl.nodeName))
+		if err != nil {
+			return fmt.Errorf("error getting node from cache %q: %v", kl.nodeName, err)
```

Comment:
an error fetching from cache should fall back to the current path, not error

Reply:
We have `nodeStatusUpdateRetry = 5` retry on higher loop.

Reply:
No, don't waste an attempt on a faulty cache, fall back to a live get immediately

Reply:
ic